packages/nextjs/scripts/database/sync_bg_locations.ts

```diff
@@ -0,0 +1,65 @@
+import { client, db } from ".";
+import { lower, users } from "../../services/database/config/schema";
+import { fetchAndCacheBGBuildersData } from "../utils/fetchAndCacheBGBuildersData";
+import { eq, inArray } from "drizzle-orm";
+import { NeonHttpDatabase } from "drizzle-orm/neon-http";
+import { Pool } from "pg";
+
+async function main() {
+  const buidlGuidlUsers = await fetchAndCacheBGBuildersData();
+
+  const buidlGuidlUserAddresses = buidlGuidlUsers.map(builder => builder.id.toLowerCase());
+
+  const existingSREV2Users = await db.query.users.findMany({
+    where: inArray(lower(users.userAddress), buidlGuidlUserAddresses),
+    columns: {
+      userAddress: true,
+      location: true,
+    },
+  });
+
+  const userMap = new Map(existingSREV2Users.map(user => [user.userAddress.toLowerCase(), user]));
+
+  const updates: any[] = [];
+  let updatedCount = 0;
+
+  for (const builder of buidlGuidlUsers) {
+    const builderAddress = builder.id?.toLowerCase();
+    const builderLocation = builder.location;
+    if (!builderAddress || !builderLocation) continue;
+
+    const user = userMap.get(builderAddress);
+
+    if (user && !user.location) {
+      const updateQuery = db
+        .update(users)
+        .set({ location: builderLocation, updatedAt: new Date() })
+        .where(eq(lower(users.userAddress), builderAddress));
+
+      if (db instanceof NeonHttpDatabase) {
+        updates.push(updateQuery);
+      } else {
+        await updateQuery;
+        updatedCount++;
+      }
+    }
+  }
+
+  if (db instanceof NeonHttpDatabase && updates.length > 0) {
+    await db.batch(updates as any);
+    updatedCount += updates.length;
+    console.log(`Batch updated ${updates.length} users.`);
+  }
```

Comment:
This works nciely, tried attaching to personal neon account and : 
![image](https://github.com/user-attachments/assets/3d472e6d-0bdf-4fe5-a6c3-f1307f97e7b5)